packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/foundation.dart';
+
+import 'package:flutter/cupertino.dart';
+
+const EdgeInsets _kHorizontalItemPadding =
+    const EdgeInsets.symmetric(horizontal: 16.0, vertical: 0.0);
+
+const Radius defaultCornerRadius = const Radius.circular(3.0);
+
+/// An iOS-style segmented control bar
+///
+/// Displays the provided [options] widgets in a horizontal list.
+/// Exactly one [options] widget is always selected. By default,
+/// the leftmost widget is selected.
+///
+/// All widgets have the same width.
+///
+/// All widgets provided should have unique Key property.
+/// These keys will be used to identify the currently selected
+/// widget.
```

Comment:
You might suggest the "typical" case: that they will create a [ValueKey<T>] with the value that the use to differentiate the widgets.